### ISSUE TITLE
Decryption failed - key packet was already decrypted

### ISSUE THREAD

**Issue by DenBond7** [FlowCrypt/flowcrypt-android#784]
it happened on the change a passphrase screen (only once for all time)

```
[SUMMARY]
com.flowcrypt.email.util.exception.ManualHandledException: Handled manually:
com.flowcrypt.email.util.exception.NodeException: Error :Error: Decryption failed - key packet was already decrypted

[STACK_TRACE]
com.flowcrypt.email.util.exception.ManualHandledException: Handled manually:
        at com.flowcrypt.email.util.exception.ExceptionUtil$Companion.handleError(ExceptionUtil.kt:30)
        at com.flowcrypt.email.ui.loader.ChangePassPhraseAsyncTaskLoader.loadInBackground(ChangePassPhraseAsyncTaskLoader.kt:85)
        at com.flowcrypt.email.ui.loader.ChangePassPhraseAsyncTaskLoader.loadInBackground(ChangePassPhraseAsyncTaskLoader.kt:33)
        at androidx.loader.content.AsyncTaskLoader.onLoadInBackground(AsyncTaskLoader.java:307)
        at androidx.loader.content.AsyncTaskLoader$LoadTask.doInBackground(AsyncTaskLoader.java:60)
        at androidx.loader.content.AsyncTaskLoader$LoadTask.doInBackground(AsyncTaskLoader.java:48)
        at androidx.loader.content.ModernAsyncTask$2.call(ModernAsyncTask.java:141)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:764)
Caused by: com.flowcrypt.email.util.exception.NodeException: Error :Error: Decryption failed - key packet was already decrypted
Stack :Error: Decryption failed - key packet was already decrypted
    at Object.decrypt ([eval]:76070:17)
    at Endpoints.decryptKey ([eval]:86885:31)
    at process._tickCallback (example.file:68:7)
Type :null
        at com.flowcrypt.email.api.retrofit.node.NodeCallsExecutor$Companion.checkResult(NodeCallsExecutor.kt:189)
        at com.flowcrypt.email.api.retrofit.node.NodeCallsExecutor$Companion.decryptKey(NodeCallsExecutor.kt:112)
        at com.flowcrypt.email.api.retrofit.node.NodeCallsExecutor$Companion.decryptKey(NodeCallsExecutor.kt:92)
        at com.flowcrypt.email.ui.loader.ChangePassPhraseAsyncTaskLoader.getModifiedNodeKeyDetails(ChangePassPhraseAsyncTaskLoader.kt:108)
        at com.flowcrypt.email.ui.loader.ChangePassPhraseAsyncTaskLoader.loadInBackground(ChangePassPhraseAsyncTaskLoader.kt:66)
        ... 9 more
com.flowcrypt.email.util.exception.NodeException: Error :Error: Decryption failed - key packet was already decrypted
Stack :Error: Decryption failed - key packet was already decrypted
    at Object.decrypt ([eval]:76070:17)
    at Endpoints.decryptKey ([eval]:86885:31)
    at process._tickCallback (example.file:68:7)
Type :null
        at com.flowcrypt.email.api.retrofit.node.NodeCallsExecutor$Companion.checkResult(NodeCallsExecutor.kt:189)
        at com.flowcrypt.email.api.retrofit.node.NodeCallsExecutor$Companion.decryptKey(NodeCallsExecutor.kt:112)
        at com.flowcrypt.email.api.retrofit.node.NodeCallsExecutor$Companion.decryptKey(NodeCallsExecutor.kt:92)
        at com.flowcrypt.email.ui.loader.ChangePassPhraseAsyncTaskLoader.getModifiedNodeKeyDetails(ChangePassPhraseAsyncTaskLoader.kt:108)
        at com.flowcrypt.email.ui.loader.ChangePassPhraseAsyncTaskLoader.loadInBackground(ChangePassPhraseAsyncTaskLoader.kt:66)
        at com.flowcrypt.email.ui.loader.ChangePassPhraseAsyncTaskLoader.loadInBackground(ChangePassPhraseAsyncTaskLoader.kt:33)
        at androidx.loader.content.AsyncTaskLoader.onLoadInBackground(AsyncTaskLoader.java:307)
        at androidx.loader.content.AsyncTaskLoader$LoadTask.doInBackground(AsyncTaskLoader.java:60)
        at androidx.loader.content.AsyncTaskLoader$LoadTask.doInBackground(AsyncTaskLoader.java:48)
        at androidx.loader.content.ModernAsyncTask$2.call(ModernAsyncTask.java:141)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:764)
```

ref https://mail.google.com/mail/u/0/#search/%22Decryption+failed+-+key+packet+was+already+decrypted%22

**Comment by tomholub** [comment]
switching to `k.isFullyDecrypted` and `k.isFullyEncrypted` should fix this. Android is still using old `k.isDecrypted`

**Comment by DenBond7** [comment]
#708

**Comment by DenBond7** [comment]
I've moved this issue to the same milestone

**Comment by tomholub** [comment]
superseded by #976
